compiler/rustc_trait_selection/src/traits/engine.rs

```diff
@@ -182,6 +195,11 @@ impl<'a, 'tcx> ObligationCtxt<'a, 'tcx> {
         self.engine.borrow_mut().select_all_or_error(self.infcx)
     }
 
+    #[must_use]
+    pub fn pending_obligations(&self) -> Vec<PredicateObligation<'tcx>> {
```

Comment:
This could also consume the `ocx`, since registering more goals (or doing anything else) will be ignored.